js/projection-chain/odata.js

```diff
@@ -0,0 +1,53 @@
+import Promise from 'bluebird';
+import _ from 'underscore';
+import $ from 'jquery';
+import Backbone from 'backbone';
+
+function odata (state, {
+	verb = 'get',
+	url,
+	skip,
+	take,
+	filter,
+	orderby = [],
+	select = [],
+} = {}) {
+
+	var url = _.isFunction(url) ? url() : url;
+	var op = {
+		url: url,
+		$format: 'json',
+		$count: true,
+	};
+
+	if (take) {
+		op.$top = take;
+	}
+
+	if (skip) {
+		op.$skip = skip;
+	}
+
+	if (_.size(orderby)) {
+		var col = _.first(orderby);
+		var key = _.keys(col)[0];
+		var dir = col[key];
+
+		op.$orderby = key + ' ' + (dir > 0 ? 'asc' : 'desc');
+	}
+
+	
+	var promiseData  = new Promise(function (resolve, reject) {
```

Comment:
We can be consistent, use `p$state`.
The concepts of `state` and `props` come from React.JS